hibp.js

```diff
@@ -65,6 +63,25 @@ const HIBP = {
     return await this._throttledGot(url, reqOptions);
   },
 
+  matchFluentID(dataCategory) {
+    dataCategory = dataCategory.replace(/[^-a-z0-9]/ig,"-").toLowerCase();
+    dataCategory = dataCategory.replace(/-+/g, "-");
+    dataCategory = dataCategory.replace(/-$/g, "");
+    return dataCategory;
+  },
+
+  formatDataClassesArray(dataCategories) {
+    const formattedArray = [];
+    if (dataCategories.constructor === Array) {
+      dataCategories.forEach(category => {
+        formattedArray.push(this.matchFluentID(category));
+      });
+      return formattedArray;
+    } else {
+      return formattedArray.push(this.matchFluentID(dataCategories));
```

Comment:
This looks suspicious. 
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/push

Looks like this would return the length of the array (always 1 in this case), whereas the `if` above is returning an actual array.

🎤 👇